src/storage/exec/ScanNode.h

```diff
@@ -0,0 +1,208 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include "common/base/Base.h"
+#include "storage/exec/GetPropNode.h"
+
+namespace nebula {
+namespace storage {
+
+using Cursor = std::string;
+
+// Node to scan vertices of one partition
+class ScanVertexPropNode : public QueryNode<Cursor> {
+ public:
+  using RelNode<Cursor>::doExecute;
+
+  explicit ScanVertexPropNode(RuntimeContext* context,
+                              std::vector<std::unique_ptr<TagNode>> tagNodes,
+                              bool enableReadFollower,
+                              int64_t limit,
+                              std::unordered_map<PartitionID, cpp2::ScanCursor>* cursors,
+                              nebula::DataSet* resultDataSet)
+      : context_(context),
+        tagNodes_(std::move(tagNodes)),
+        enableReadFollower_(enableReadFollower),
+        limit_(limit),
+        cursors_(cursors),
+        resultDataSet_(resultDataSet) {
+    name_ = "ScanVertexPropNode";
+    std::vector<TagNode*> tags;
```

Comment:
I don't quite understand, would you explain more about why we need the `GetTagPropNode` here?
Since you have already scan data by rocksdb iterator (line 60), why we use get again(line 80)?

Reply:
1. line 60 just get one tag key, but we support scan multiple tags.
2. behave same with GetProp interface, so I implment ScanVertexPropNode by wrap the GetTagPropNode.

Reply:
For second, the graph will process the result of ScanVertex and GetProp in same code.

Reply:
I got your point, some of them looks weird:
1. We don't support multiple tags/edges for now, see `VertexProp` and `EdgeProp` in the `ScanVertexRequest` and `ScanEdgeRequest`. We can't support them unless we separate data into different DataSet according to tag/edgeType.
2. The return format could be unified, but I don't suggest to implement like now, because the extra `get` in rocksdb is really a waste.

Reply:
1. Sorry it's my fault, #3106 require scan multiple tags, I will add it later like `GetProp`.
2. Yes, I will try to optimize it.